crates/cheatcodes/src/evm.rs

```diff
@@ -355,13 +350,7 @@ impl Cheatcode for getBlobhashesCall {
 impl Cheatcode for rollCall {
     fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
         let Self { newHeight } = self;
-        if ccx.state.use_zk_vm {
-            foundry_zksync_core::cheatcodes::roll(*newHeight, ccx.ecx);
-            return Ok(Default::default())
-        }
-
-        ccx.ecx.env.block.number = *newHeight;
-        Ok(Default::default())
+        ccx.with_strategy(|strategy, ccx| strategy.cheatcode_roll(ccx, *newHeight))
```

Comment:
So say upstream changes the implementation of roll cheatcode, we will get a merge conflict here then have to move the new implementation to the EvmCheatcodeInspectorStrategy ? I do fear that this procedure might not be as trivial and error prone as it seems at first glance. Unless of course the strategy approach is implemented upstream.